core/eds_test.go

```diff
@@ -28,16 +27,19 @@ func TestTrulyEmptySquare(t *testing.T) {
 }
 
 // TestNonEmptySquareWithZeroTxs tests that a non-empty square with no
-// transactions or blobs computes the correct data root (not the minimum DAH).
+// transactions or blobs computes the correct data root the minimum DAH.
+// Technically, this block data is invalid because the construction of the
+// square is deterministic, and the rules which dictate the square size do not
+// allow for empty block data. However, should that ever occur, we need to to
+// ensure that the correct data root is generated.
 func TestNonEmptySquareWithZeroTxs(t *testing.T) {
 	data := types.Data{
 		Txs:        []types.Tx{},
```

Comment:
yes that's a good point, we should. however the only way to know that is to calculate the square to check :grimacing: 

we will document that the square size in the block data should never be used, while it does get checked by all consensus nodes, the only reason we still have it is so that tendermint can have easy access to it instead of having to ask the application to calculate the square and find out.

created an issue https://github.com/celestiaorg/celestia-core/issues/1015